### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by kevindlewis23**
Codecov needs to be setup.  Get a token from [here](https://app.codecov.io/gh/Hexrd) and setup for hexrd.  Must be added as a workflow secret in settings.

Once it is, this PR adds a code coverage workflow and a status badge to the readme.

Maybe refactor to add the code coverage workflow to be part of the test workflow.  I did it separately because the test workflow runs on many OS's and that is unnecessary for coverage checks.  Also codecov needs an editable install of hexrd, and while I assume that doesn't change anything, the actual test cases should be as close to production as possible.

Fixes: #571

### CODE REVIEW

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Run tests
+        env:
+            HEXRD_EXAMPLE_REPO_PATH: ${{ github.workspace }}/examples
+        run: |
+            pytest --cov hexrd --cov-report xml:coverage.xml tests
+        working-directory: hexrd
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (psavery) [inline .github/workflows/codecov.yml:42]: This token has been added.

**Review by psavery — COMMENTED** [review]
What we have done on other projects is to include the codecov logic within the tests workflow, but only run it on one operating system (for instance, Linux). For example, for the `Upload coverage to Codecov` action, you can say `if: ${{ matrix.config.name == 'Linux'}}`, and there are things you can do to add the extra codecov arguments as well for only Linux (and not the other operating systems).

However, I'm not sure what to do about the editable install (`-e`) part. I guess we can keep this as a separate workflow if we want the main tests to not perform an editable install. The only issue is that, if we make workflow changes to the `test.yml`, they won't be reflected here. But we might need to just work with that anyways. What do you think?

**Comment by kevindlewis23** [comment]
> However, I'm not sure what to do about the editable install (`-e`) part. I guess we can keep this as a separate workflow if we want the main tests to not perform an editable install. The only issue is that, if we make workflow changes to the `test.yml`, they won't be reflected here. But we might need to just work with that anyways. What do you think?

I think it makes more sense to move it actually.  I'm going to do that

**Review by psavery — APPROVED** [review]
Looks good to me

**Pull request merged**